src/app/BufferedReadCallback.cpp

```diff
@@ -117,7 +117,8 @@ CHIP_ERROR BufferedReadCallback::BufferListItem(TLV::TLVReader & reader)
     // TLV element. Since the tag can vary in size, for now, let's just do the safe thing. In the future, if this is a problem,
     // we can improve this.
     //
-    handle = System::PacketBufferHandle::New(chip::app::kMaxSecureSduLengthBytes);
+    const size_t bufSize = mAllowLargePayload ? chip::app::kMaxLargeSecureSduLengthBytes : chip::app::kMaxSecureSduLengthBytes;
+    handle               = System::PacketBufferHandle::New(bufSize);
     VerifyOrReturnError(!handle.IsNull(), CHIP_ERROR_NO_MEMORY);
 
 #if INET_CONFIG_ENABLE_TCP_ENDPOINT
```

Comment:
Can you remove the line below that I added? (assuming they're not necessary)

This is a change I made in https://github.com/project-chip/connectedhomeip/pull/40552 thinking it'd fix the problem but it didn't

<img width="1004" height="346" alt="image" src="https://github.com/user-attachments/assets/352cbc6a-f8e3-42d6-8200-d3b61fba2e7c" />